server/src/internalClusterTest/java/org/elasticsearch/blocks/SimpleBlocksIT.java

```diff
@@ -322,31 +324,21 @@ public void testConcurrentAddBlock() throws InterruptedException {
             IntStream.range(0, nbDocs).mapToObj(i -> prepareIndex(indexName).setId(String.valueOf(i)).setSource("num", i)).collect(toList())
         );
         ensureYellowAndNoInitializingShards(indexName);
-
-        final CountDownLatch startClosing = new CountDownLatch(1);
-        final Thread[] threads = new Thread[randomIntBetween(2, 5)];
-
         final APIBlock block = randomAddableBlock();
 
+        final int threadCount = randomIntBetween(2, 5);
+        final CyclicBarrier barrier = new CyclicBarrier(threadCount);
```

Comment:
It seems so common to block all the threads at startup like this, and benign to add it to the few places where we don't - maybe we should move this into the util too?

Reply:
I think this might needlessly slow down the cluster start/restart path, especially on beefy machines with many cores like my personal workstation :) I'd rather not do this I think. We could add another utility method around this one for it though in a follow up?

Reply:
:+1: ok